### PR TITLE
Fix query frontend crash after multiple models added

### LINKED ISSUE #149: Deploy model causes docker container to exit
To reproduce this

1. Deploy any model with version no as 1
2. Try the `/predict` api to do a simple call
3. Deploy version 2 of the same model
3. Try `/predict`

query_frontend container exits with code 139

````
clipper/query_frontend:latest        "/clipper/release/..."   38 minutes ago      Exited (139) 26 seconds ago
````

This is the container log before exiting

````
[15:17:44.212][info]     [CLIPPER] Adding new container - model: captcha_predict_model, version: 6, ID: 1, input_type: strings
[15:17:44.212][info]  [TASKEXE...] Created queue for new model: captcha_predict_model : 6
[15:17:44.224][info]       [REDIS] Successfully issued command "HMSET captcha_predict_model,6,0 model_id captcha_predict_model:6 model_name captcha_predict_model model_version 6 model_replica_id 0 zmq_connection_id 1 batch_size 1 input_type strings"
[15:17:46.600][info]         [RPC] Found message to receive
[15:17:49.217][info]         [RPC] Found message to receive
[15:17:51.609][info]         [RPC] Found message to receive
[15:17:52.517][info]       [REDIS] Successfully issued command "GET captcha_predict:0:0"
[15:17:52.517][info]  [QUERYPR...] Found 1 tasks
[15:17:52.525][info]         [RPC] Found message to receive
````

Is there any other log i can look at whats happening. I am using the NoopContainer

### PR DESCRIPTION
**Pull request by dcrankshaw**
Fixes #149. The issue was a mixup between the __replica_id__ and __container_id__ assigned to a container.

### CODE REVIEW

File: src/libclipper/include/clipper/task_executor.hpp [hunk 1]
```diff
@@ -488,12 +490,18 @@ class TaskExecutor {
       boost::optional<ModelMetrics> cur_model_metric) {
     std::shared_ptr<ModelContainer> processing_container =
         active_containers_->get_model_replica(completed_msg.model_,
-                                              completed_msg.container_id_);
+                                              completed_msg.replica_id_);
```
Comment (dcrankshaw) [inline src/libclipper/include/clipper/task_executor.hpp:493]: This was the critical change. Active containers is a map of maps that keys containers by <VersionedModelID, replica_id> but we were looking them up by <VersionedModelID, container_id>. As soon as replica_id and container_id diverged, the container wouldn't be found and a nullptr was returned. We also failed to check if `processing_container` was null, so we tried to access a null ptr and caused a segfault.

File: src/libclipper/src/containers.cpp [hunk 2]
```diff
@@ -90,24 +98,40 @@ ActiveContainers::ActiveContainers()
...
-      std::make_shared<ModelContainer>(model, connection_id, input_type);
+  auto new_container = std::make_shared<ModelContainer>(model, connection_id,
+                                                        replica_id, input_type);
   auto entry = containers_[new_container->model_];
   entry.emplace(replica_id, new_container);
   containers_[new_container->model_] = entry;
   assert(containers_[new_container->model_].size() > 0);
+  std::stringstream log_msg;
+  log_msg << "\nActive containers:\n";
+  // log_info(LOGGING_TAG_CONTAINERS, "All active containers:");
```
Comment (Corey-Zumar) [inline src/libclipper/src/containers.cpp]: Remove commented code

**Review by Corey-Zumar — APPROVED** [review]
LGTM. Sorry for causing this issue in the first place. Just one comment for code cleanup.

File: src/libclipper/src/containers.cpp (hunk 2 above)
Comment (dcrankshaw) [reply src/libclipper/src/containers.cpp]: Done

**Comment by dcrankshaw** [comment]
No worries. I'm writing some integration tests that should help us catch things like this in the future.

**Pull request merged**